### PR TITLE
Add support for scraping tags in radarr

### PR DESCRIPTION
**Pull request by aneurinprice**
This change adds a new metric, the number of movies in a tag
No known limitations

Proof of functionality:

```
radarr_movie_tag_total{tag="backed_up",url="http://<server>:7878"} 437
```

**Description of the change**

Add a new metric which scrapes radarr tags and the movies inside them

**Benefits**

Additional metrics are always nice?

**Possible drawbacks**

Any drawbacks that come from adding a new metric.

### CODE REVIEW

File: internal/arr/collector/radarr.go [hunk 1]
```diff
@@ -72,6 +73,12 @@ func NewRadarrCollector(c *config.ArrConfig) *radarrCollector {
 			[]string{"quality"},
 			prometheus.Labels{"url": c.URL},
 		),
+		movieTags: prometheus.NewDesc(
```
Comment (rtrox) [inline internal/arr/collector/radarr.go:76]: Can we match the naming convention of the other metrics?

Something like movieTagsMetric?

File: internal/arr/model/radarr.go [hunk 2]
```diff
@@ -16,3 +16,9 @@ type Movie []struct {
 	} `json:"movieFile"`
 	QualityProfileID int `json:"qualityProfileId"`
 }
+
+type TagObjects []struct {
```
Comment (rtrox) [inline internal/arr/model/radarr.go:20]: This is specific to radarr, let's call this TagMovies

File: internal/arr/collector/radarr.go [hunk 3]
```diff
@@ -72,6 +73,12 @@ func NewRadarrCollector(c *config.ArrConfig) *radarrCollector {
...
+			"radarr_movie_tag_total",
+			"Tags by total",
```
Comment (rtrox) [inline internal/arr/collector/radarr.go:78]: Same here, can we match the other metrics?

"Total number of downloaded movies by tag"

**Review by rtrox — CHANGES_REQUESTED** [review]
A few nits on to keep naming/labeling consistent, but the core logic lgtm! Glad to approve and merge once resolved

File: internal/arr/collector/radarr.go (hunk 3 above)
Comment (aneurinprice) [reply internal/arr/collector/radarr.go]: Updated.

File: internal/arr/model/radarr.go (hunk 2 above)
Comment (aneurinprice) [reply internal/arr/model/radarr.go]: Updated.

File: internal/arr/collector/radarr.go (hunk 1 above)
Comment (aneurinprice) [reply internal/arr/collector/radarr.go]: Updated.

File: internal/arr/model/radarr.go [hunk 4]
```diff
@@ -16,3 +16,9 @@ type Movie []struct {
 	} `json:"movieFile"`
 	QualityProfileID int `json:"qualityProfileId"`
 }
+
+type TagMovies []struct {
+	ID	int	`json:"id"`
+	Label string `json:"label"`
+	MovieIds []int `json:"movieIds"`
+}
```
Comment (onedr0p) [inline internal/arr/model/radarr.go:24]:
```suggestion
}

```

File: internal/arr/collector/radarr.go [hunk 5]
```diff
@@ -138,6 +149,27 @@ func (collector *radarrCollector) Collect(ch chan<- prometheus.Metric) {
 			fileSize += s.MovieFile.Size
 		}
 	}
+
+	tagObjects := model.TagMovies{}
```
Comment (onedr0p) [inline internal/arr/collector/radarr.go:153]: I think to use `tagMovies` here as well or `movieTags` ?

**Review by onedr0p — CHANGES_REQUESTED** [review]

File: internal/arr/collector/radarr.go (hunk 5 above)
Comment (onedr0p) [reply internal/arr/collector/radarr.go:153]: Nevermind, this makes sense!

**Review by onedr0p — APPROVED** [review]
Looks good from here, @rtrox I'll wait for your sign off too.

**Review by rtrox — APPROVED** [review]
Lgtm too! Thanks for the quick follow-up.

**Pull request merged**